### DISCUSSION TITLE
Hey Rector Team! Can Rector generate custom rules from classmap extraction using for example Rector FileProcessor?

### DISCUSSION THREAD

**Discussion by alexandreelise** [rectorphp/rector#7426] [Q&A] [answered]
Hi,

I am doing manually the task of creating a custom rector rule for each class alias of my favourite framework ('cough' Joomla) so it can do a rename for static calls 

e.g

```diff

- JRoute::
+ \Joomla\Router\Route::

```
But I pretty think there is an automated way of generating all those rules using Rector to generate a custom Rector rule. Kinda inception. I am aware of rector generate but rather want to parse a class alias php file than generate each custom rule with the corresponding code in it. I know it's doable but is it the correct approach to do it. Can you shed a light on this one please @TomasVotruba

**Comment by TomasVotruba** [comment, marked as answer]
Hi,

what is your goal exactly? Rename class?

**Reply by alexandreelise** [reply]
I wanna achieve what is in the code sample. I don't know if it is just class renaming.

**Reply by TomasVotruba** [reply]
Seems like renaming `JRoute` class to `Joomla\Router\Route`, unless there is more code that I don't see :)

In that case, you can use rename class rule :+1:

**Reply by alexandreelise** [reply]
Yep actually I know rename class rule but I was hesitating of using it. Funny enough, I could have written a simple case and try it with --dry-run flag of rector to see if it works as expected. That's what I do now for my custom rules since there is a phpstan phpstorm stubs already declared bug on my setup I use rector directly with the paths configured to scan only Fixtures dir that way I can pinpoint if it works for small reproductible cases with --dry-run. When it's ok. I add then execute the custom rector rule to the real code. It works like a charm.

**Reply by alexandreelise** [reply]
Hi @TomasVotruba ,

I prepared the class alias associative array, did an array_merge with another classmap, used it with RenameClassRector rule with configuration and it worked flawlessly.
Who wants to manually edit 1000+ files? Anyone? Not me! Thanks again Tomas, Thanks Rector Team, Big thanks to Rector!
